src/main/java/com/cognite/beam/io/fn/IOBaseFn.java

```diff
@@ -69,16 +69,21 @@ protected CogniteClient getClient(ProjectConfig projectConfig, ConfigBase config
 
             if (configHasClientCredentials(projectConfig)) {
                 // client credentials take precedence over api key
-                client = CogniteClient.ofClientCredentials(projectConfig.getClientId().get(),
-                        projectConfig.getClientSecret().get(), new URL(projectConfig.getTokenUrl().get()))
+                client = CogniteClient.ofClientCredentials(
+                        projectConfig.getProject().get(),
+                        projectConfig.getClientId().get(),
+                        projectConfig.getClientSecret().get(),
+                                new URL(projectConfig.getTokenUrl().get()))
                         .withBaseUrl(projectConfig.getHost().get())
                         .withClientConfig(clientConfig);
                 // set custom auth scopes if they are configured in the ProjectConfig object.
                 if (null != projectConfig.getAuthScopes() && projectConfig.getAuthScopes().isAccessible()) {
                     client = client.withScopes(projectConfig.getAuthScopes().get());
                 }
             } else if (configHasApiKey(projectConfig)) {
```

Comment:
We have recently removed the API key option from the Java client, so I think it is safe to remove from here as well.